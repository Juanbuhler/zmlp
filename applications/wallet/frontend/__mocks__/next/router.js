/**
 * useRouter
 */

let mockUseRouter = {
  query: {
    action: '',
  },
}

export const __setUseRouter = data => {
  mockUseRouter = data
}

export const useRouter = () => mockUseRouter

/**
 * push
 */

let mockPushFunction = () => {}

export const __setMockPushFunction = fn => {
  mockPushFunction = fn
}

/**
 * on
 */

let mockOnFunction = () => {}

export const __setMockOnFunction = fn => {
  mockOnFunction = fn
}

/**
 * off
 */

let mockOffFunction = () => {}

export const __setMockOffFunction = fn => {
  mockOffFunction = fn
}

/**
 * Router
 */

const Router = {
  push: (...args) => {
    mockPushFunction(...args)
  },
  events: {
    on: (...args) => {
      mockOnFunction(...args)
    },
    off: (...args) => {
      mockOffFunction(...args)
    },
  },
}

export default Router
