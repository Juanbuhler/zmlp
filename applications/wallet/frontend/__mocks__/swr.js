import { createElement } from 'react'

const noop = () => () => {}

/**
 * <SWRConfig />
 */

export const SWRConfig = ({ children, ...rest }) =>
  createElement('SWRConfig', rest, children)

/**
 * cache
 */

export const { cache } = jest.requireActual('swr')

/**
 * useSWR
 */

let mockUseSWRResponse = { revalidate: noop, mutate: noop }

export const __setMockUseSWRResponse = (data) => {
  mockUseSWRResponse = { revalidate: noop, mutate: noop, ...data }
}

const useSWR = () => {
  return mockUseSWRResponse
}

export default useSWR
