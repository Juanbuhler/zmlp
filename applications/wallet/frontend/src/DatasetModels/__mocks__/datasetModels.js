const datasetModels = {
  count: 1,
  results: [
    {
      id: '313e28b1-9f0e-1595-a6c0-ea38f4c81474',
      name: 'tensorflow-quality',
      type: 'TF_CLASSIFIER',
    },
  ],
}

export default datasetModels
