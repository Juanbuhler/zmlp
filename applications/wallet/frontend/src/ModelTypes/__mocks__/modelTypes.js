const modelTypes = {
  results: [
    {
      name: 'ZVI_KNN_CLASSIFIER',
      label: 'Sci-kit Learn KNN Classifier',
      description:
        'Classify images or documents using a KNN classifier.  This type of model generates a single prediction which can be used to quickly organize assets into general groups.The KNN classifier works with just a single image and label.',
      objective: 'Label Detection',
      provider: 'Zorroa',
      deployOnTrainingSet: true,
      minConcepts: 0,
      minExamples: 0,
    },
    {
      name: 'ZVI_LABEL_DETECTION',
      label: 'Tensorflow CNN Classifier',
      description:
        'Classify images or documents using a custom strained CNN deep learning algorithm.  This type of modelgenerates multiple predictions and can be trained to identify very specific features. The label detection classifier requires at least 2 concepts with 10 labeled images each.',
      objective: 'Label Detection',
      provider: 'Zorroa',
      deployOnTrainingSet: false,
      minConcepts: 2,
      minExamples: 10,
    },
    {
      name: 'TF2_IMAGE_CLASSIFIER',
      label: 'Tensorflow2 Keras Image Classifier',
      description: 'Upload a Tensorflow model to use for image classification.',
      objective: 'Label Detection',
      provider: 'Google',
      deployOnTrainingSet: true,
      minConcepts: 0,
      minExamples: 0,
    },
    {
      name: 'GCP_LABEL_DETECTION',
      label: 'Google AutoML Classifier',
      description: 'Utilize Google AutoML to train an image classifier.',
      objective: 'Label Detection',
      provider: 'Google',
      deployOnTrainingSet: true,
      minConcepts: 2,
      minExamples: 10,
    },
  ],
}

export default modelTypes
