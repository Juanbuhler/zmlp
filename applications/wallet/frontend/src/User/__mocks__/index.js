import { createContext, createElement } from 'react'

import user from './user'

export const noop = () => {}

export const UserContext = createContext({
  user,
  googleAuth: {},
  setGoogleAuth: noop,
})

const User = ({ children, ...rest }) => createElement('User', rest, children)

export default User
