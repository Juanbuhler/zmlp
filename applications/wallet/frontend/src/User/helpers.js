export const noop = () => {}

export const meFetcher =
  typeof window === 'undefined'
    ? noop
    : async (url) => {
        try {
          const response = await fetch(url)
          if (response.status >= 400) throw response
          return response.json()
        } catch (error) {
          return {}
        }
      }
