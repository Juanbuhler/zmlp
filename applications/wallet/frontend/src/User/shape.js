import PropTypes from 'prop-types'

const userShape = {
  id: PropTypes.number.isRequired,
  username: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  organizations: PropTypes.arrayOf(PropTypes.string).isRequired,
}

export default userShape
