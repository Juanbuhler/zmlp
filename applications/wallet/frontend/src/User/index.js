import { createContext, useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import * as Sentry from '@sentry/browser'

import { getUser, setUser } from './helpers'

export const noop = () => {}

export const UserContext = createContext({
  user: {},
  setUser: noop,
  googleAuth: {},
  setGoogleAuth: noop,
})

const User = ({ initialUser, children }) => {
  const [user, setStateUser] = useState(initialUser)
  const [hasLocalStorageLoaded, setHasLocalStorageLoaded] = useState(false)
  const [googleAuth, setGoogleAuth] = useState({ signIn: noop, signOut: noop })

  useEffect(() => {
    if (initialUser.id || hasLocalStorageLoaded) return

    const storedUser = getUser()

    setStateUser(storedUser)

    setHasLocalStorageLoaded(true)

    /* istanbul ignore next */
    Sentry.configureScope(scope => {
      scope.setUser(storedUser)
    })
  }, [initialUser, hasLocalStorageLoaded, user, setStateUser])

  if (!initialUser.id && !hasLocalStorageLoaded) return null

  return (
    <UserContext.Provider
      value={{
        user,
        setUser: setUser({ setStateUser, user }),
        googleAuth,
        setGoogleAuth,
      }}>
      {children}
    </UserContext.Provider>
  )
}

User.propTypes = {
  initialUser: PropTypes.shape({
    id: PropTypes.number,
    email: PropTypes.string,
  }).isRequired,
  children: PropTypes.node.isRequired,
}

export default User
