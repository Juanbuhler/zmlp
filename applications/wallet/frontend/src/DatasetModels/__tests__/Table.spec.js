import TestRenderer, { act } from 'react-test-renderer'

import mockUser from '../../User/__mocks__/user'
import datasetModels from '../__mocks__/datasetModels'
import modelTypes from '../../ModelTypes/__mocks__/modelTypes'

import User from '../../User'

import DatasetModelsTable from '../Table'

const PROJECT_ID = '76917058-b147-4556-987a-0a0f11e46d9b'
const DATASET_ID = '621bf775-89d9-1244-9596-d6df43f1ede5'

describe('<DatasetModelsTable />', () => {
  it('should render properly without models', async () => {
    require('next/router').__setUseRouter({
      pathname: '/[projectId]/datasets/[datasetId]/models',
      query: { projectId: PROJECT_ID, datasetId: DATASET_ID },
    })

    require('swr').__setMockUseSWRResponse({ data: {} })

    const component = TestRenderer.create(
      <User initialUser={mockUser}>
        <DatasetModelsTable
          projectId={PROJECT_ID}
          datasetId={DATASET_ID}
          modelTypes={modelTypes.results}
        />
      </User>,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render properly with models', async () => {
    require('next/router').__setUseRouter({
      pathname: '/[projectId]/datasets/[datasetId]/models',
      query: { projectId: PROJECT_ID, datasetId: DATASET_ID },
    })

    require('swr').__setMockUseSWRResponse({ data: datasetModels })

    const component = TestRenderer.create(
      <User initialUser={mockUser}>
        <DatasetModelsTable
          projectId={PROJECT_ID}
          datasetId={DATASET_ID}
          modelTypes={modelTypes.results}
        />
      </User>,
    )

    // Open Menu
    act(() => {
      component.root
        .findByProps({ 'aria-label': 'Toggle Actions Menu' })
        .props.onClick()
    })

    expect(component.toJSON()).toMatchSnapshot()
  })
})
