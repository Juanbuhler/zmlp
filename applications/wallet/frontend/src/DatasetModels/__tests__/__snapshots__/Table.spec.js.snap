// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatasetModelsTable /> should render properly with an unknown model type 1`] = `
.emotion-0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-bottom: 16px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-3 {
  color: #b3b3b3;
  font-size: 15px;
  line-height: 19px;
  font-weight: 400;
}

.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.emotion-6 {
  height: auto;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
  white-space: nowrap;
}

.emotion-6 tr {
  background-color: #252525;
}

.emotion-6 tr:nth-of-type(2n) {
  background-color: #2e2e2e;
}

.emotion-6 tr:hover {
  background-color: #4a4a4a;
  box-shadow: 0 0 5px 0 rgba(0, 0, 0, 0.5);
}

.emotion-6 tr:hover .actions {
  color: #b3b3b3;
}

.emotion-6 tr:hover td {
  border: 1px solid #808080;
  border-left: 0;
  border-right: 0;
}

.emotion-6 tr:hover td:first-of-type {
  border-left: 1px solid #808080;
}

.emotion-6 tr:hover td:last-of-type {
  border-right: 1px solid #808080;
}

.emotion-6 td {
  max-width: calc(100vw / 2);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  color: #e0e0e0;
  padding: 8px 16px;
  border: 1px solid transparent;
  border-left: 0;
  border-right: 0;
}

.emotion-6 td:first-of-type {
  border-left: 1px solid transparent;
}

.emotion-6 td:last-of-type {
  border-right: 1px solid transparent;
  overflow: visible;
}

.emotion-7 {
  text-align: left;
  white-space: pre-line;
  vertical-align: bottom;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  color: #e0e0e0;
  background-color: #4a4a4a;
  padding: 0;
  border-bottom: 1px solid #2e2e2e;
  width: auto;
}

.emotion-7:nth-of-type(0) {
  width: 100%;
}

.emotion-7:not(:last-child) {
  border-right: 1px solid #2e2e2e;
}

.emotion-8 {
  padding: 12px 16px;
}

.emotion-11 {
  text-align: left;
  white-space: pre-line;
  vertical-align: bottom;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  color: #e0e0e0;
  background-color: #4a4a4a;
  padding: 0;
  border-bottom: 1px solid #2e2e2e;
  width: 1px;
}

.emotion-11:nth-of-type(0) {
  width: 100%;
}

.emotion-11:not(:last-child) {
  border-right: 1px solid #2e2e2e;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 12px 16px;
}

.emotion-13 {
  cursor: pointer;
}

.emotion-14:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-15 {
  position: relative;
  height: 100%;
}

.emotion-16 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  padding: 6px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #808080;
  background-color: transparent;
  margin: -6px;
}

.emotion-16:hover,
.emotion-16.focus-visible:focus {
  color: #ffffff!important;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-2"
    >
      <h3
        className="emotion-3"
      >
        Models
        : 
        1
      </h3>
    </div>
    <div
      className="emotion-4"
    >
      <div
        className="emotion-5"
      >
        <table
          className="emotion-6"
        >
          <thead>
            <tr>
              <th
                className="emotion-7"
              >
                <div
                  className="emotion-8"
                >
                  Name
                </div>
              </th>
              <th
                className="emotion-7"
              >
                <div
                  className="emotion-8"
                >
                  Type
                </div>
              </th>
              <th
                className="emotion-11"
              >
                <div
                  className="emotion-12"
                >
                  <svg
                    data-file-name="kebab.svg"
                    height={20}
                  />
                </div>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              className="emotion-13"
              onClick={[Function]}
            >
              <td>
                <a
                  className="emotion-14"
                  href="/76917058-b147-4556-987a-0a0f11e46d9b/models/313e28b1-9f0e-1595-a6c0-ea38f4c81474"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  tensorflow-quality
                </a>
              </td>
              <td>
                TF_CLASSIFIER
              </td>
              <td>
                <div
                  className="emotion-15"
                >
                  <button
                    aria-label="Toggle Actions Menu"
                    className="actions emotion-16"
                    onBlur={[Function]}
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      data-file-name="kebab.svg"
                      height={20}
                    />
                  </button>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div>
         
      </div>
      <div>
         
      </div>
    </div>
  </div>
</div>
`;

exports[`<DatasetModelsTable /> should render properly with models 1`] = `
.emotion-0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-bottom: 16px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-3 {
  color: #b3b3b3;
  font-size: 15px;
  line-height: 19px;
  font-weight: 400;
}

.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.emotion-6 {
  height: auto;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
  white-space: nowrap;
}

.emotion-6 tr {
  background-color: #252525;
}

.emotion-6 tr:nth-of-type(2n) {
  background-color: #2e2e2e;
}

.emotion-6 tr:hover {
  background-color: #4a4a4a;
  box-shadow: 0 0 5px 0 rgba(0, 0, 0, 0.5);
}

.emotion-6 tr:hover .actions {
  color: #b3b3b3;
}

.emotion-6 tr:hover td {
  border: 1px solid #808080;
  border-left: 0;
  border-right: 0;
}

.emotion-6 tr:hover td:first-of-type {
  border-left: 1px solid #808080;
}

.emotion-6 tr:hover td:last-of-type {
  border-right: 1px solid #808080;
}

.emotion-6 td {
  max-width: calc(100vw / 2);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  color: #e0e0e0;
  padding: 8px 16px;
  border: 1px solid transparent;
  border-left: 0;
  border-right: 0;
}

.emotion-6 td:first-of-type {
  border-left: 1px solid transparent;
}

.emotion-6 td:last-of-type {
  border-right: 1px solid transparent;
  overflow: visible;
}

.emotion-7 {
  text-align: left;
  white-space: pre-line;
  vertical-align: bottom;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  color: #e0e0e0;
  background-color: #4a4a4a;
  padding: 0;
  border-bottom: 1px solid #2e2e2e;
  width: auto;
}

.emotion-7:nth-of-type(0) {
  width: 100%;
}

.emotion-7:not(:last-child) {
  border-right: 1px solid #2e2e2e;
}

.emotion-8 {
  padding: 12px 16px;
}

.emotion-11 {
  text-align: left;
  white-space: pre-line;
  vertical-align: bottom;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  color: #e0e0e0;
  background-color: #4a4a4a;
  padding: 0;
  border-bottom: 1px solid #2e2e2e;
  width: 1px;
}

.emotion-11:nth-of-type(0) {
  width: 100%;
}

.emotion-11:not(:last-child) {
  border-right: 1px solid #2e2e2e;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 12px 16px;
}

.emotion-13 {
  cursor: pointer;
}

.emotion-14:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-15 {
  position: relative;
  height: 100%;
}

.emotion-16 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  padding: 6px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #808080;
  background-color: transparent;
  margin: -6px;
}

.emotion-16:hover,
.emotion-16.focus-visible:focus {
  color: #ffffff!important;
}

.emotion-17 {
  position: absolute;
  z-index: 1;
  top: 100%;
  right: 0;
  background-color: #808080;
  border-radius: 2px;
  box-shadow: 0 4px 7px 0 #000000;
  min-width: 200px;
  font-family: Roboto,Avenir,"sans-serif";
}

.emotion-17 ul {
  max-height: 50vh;
  overflow: scroll;
  list-style-type: none;
  padding: 0;
  padding-top: 4px;
  padding-bottom: 4px;
  margin: 0;
}

.emotion-17 li {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-18 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 0;
  padding: 8px 16px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 400;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
}

.emotion-18:hover,
.emotion-18.focus-visible:focus,
.emotion-18 .noop {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #4a4a4a;
}

.emotion-19 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-20 {
  -webkit-transform: rotate(135deg);
  -moz-transform: rotate(135deg);
  -ms-transform: rotate(135deg);
  transform: rotate(135deg);
  opacity: 0.5;
  margin-top: -2px;
  margin-right: -4px;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-2"
    >
      <h3
        className="emotion-3"
      >
        Models
        : 
        1
      </h3>
    </div>
    <div
      className="emotion-4"
    >
      <div
        className="emotion-5"
      >
        <table
          className="emotion-6"
        >
          <thead>
            <tr>
              <th
                className="emotion-7"
              >
                <div
                  className="emotion-8"
                >
                  Name
                </div>
              </th>
              <th
                className="emotion-7"
              >
                <div
                  className="emotion-8"
                >
                  Type
                </div>
              </th>
              <th
                className="emotion-11"
              >
                <div
                  className="emotion-12"
                >
                  <svg
                    data-file-name="kebab.svg"
                    height={20}
                  />
                </div>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              className="emotion-13"
              onClick={[Function]}
            >
              <td>
                <a
                  className="emotion-14"
                  href="/76917058-b147-4556-987a-0a0f11e46d9b/models/313e28b1-9f0e-1595-a6c0-ea38f4c81474"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  tensorflow-quality
                </a>
              </td>
              <td>
                Tensorflow Transfer Learning Classifier
              </td>
              <td>
                <div
                  className="emotion-15"
                >
                  <button
                    aria-label="Toggle Actions Menu"
                    className="actions emotion-16"
                    onBlur={[Function]}
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      data-file-name="kebab.svg"
                      height={20}
                    />
                  </button>
                  <div
                    className="emotion-17"
                  >
                    <div>
                      <ul>
                        <li>
                          <a
                            className="emotion-18"
                            href="/76917058-b147-4556-987a-0a0f11e46d9b/models/313e28b1-9f0e-1595-a6c0-ea38f4c81474"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                          >
                            <div
                              className="emotion-19"
                            >
                              View Model
                              <svg
                                className="emotion-20"
                                data-file-name="back.svg"
                                height={20}
                              />
                            </div>
                          </a>
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div>
         
      </div>
      <div>
         
      </div>
    </div>
  </div>
</div>
`;

exports[`<DatasetModelsTable /> should render properly without models 1`] = `
.emotion-0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-bottom: 16px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-3 {
  color: #b3b3b3;
  font-size: 15px;
  line-height: 19px;
  font-weight: 400;
}

.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.emotion-5 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.emotion-6 {
  height: 100%;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
  white-space: nowrap;
}

.emotion-6 tr {
  background-color: #252525;
}

.emotion-6 tr:nth-of-type(2n) {
  background-color: #2e2e2e;
}

.emotion-6 tr:hover {
  background-color: #4a4a4a;
  box-shadow: 0 0 5px 0 rgba(0, 0, 0, 0.5);
}

.emotion-6 tr:hover .actions {
  color: #b3b3b3;
}

.emotion-6 tr:hover td {
  border: 1px solid #808080;
  border-left: 0;
  border-right: 0;
}

.emotion-6 tr:hover td:first-of-type {
  border-left: 1px solid #808080;
}

.emotion-6 tr:hover td:last-of-type {
  border-right: 1px solid #808080;
}

.emotion-6 td {
  max-width: calc(100vw / 2);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  color: #e0e0e0;
  padding: 8px 16px;
  border: 1px solid transparent;
  border-left: 0;
  border-right: 0;
}

.emotion-6 td:first-of-type {
  border-left: 1px solid transparent;
}

.emotion-6 td:last-of-type {
  border-right: 1px solid transparent;
  overflow: visible;
}

.emotion-7 {
  text-align: left;
  white-space: pre-line;
  vertical-align: bottom;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  color: #e0e0e0;
  background-color: #4a4a4a;
  padding: 0;
  border-bottom: 1px solid #2e2e2e;
  width: auto;
}

.emotion-7:nth-of-type(0) {
  width: 100%;
}

.emotion-7:not(:last-child) {
  border-right: 1px solid #2e2e2e;
}

.emotion-8 {
  padding: 12px 16px;
}

.emotion-11 {
  text-align: left;
  white-space: pre-line;
  vertical-align: bottom;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  color: #e0e0e0;
  background-color: #4a4a4a;
  padding: 0;
  border-bottom: 1px solid #2e2e2e;
  width: 1px;
}

.emotion-11:nth-of-type(0) {
  width: 100%;
}

.emotion-11:not(:last-child) {
  border-right: 1px solid #2e2e2e;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 12px 16px;
}

.emotion-13 {
  height: 100%;
  pointer-events: none;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  text-align: center;
  color: #808080;
  font-size: 15px;
  line-height: 19px;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 18px;
  line-height: 21px;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-2"
    >
      <h3
        className="emotion-3"
      >
        Models
        : 
        0
      </h3>
    </div>
    <div
      className="emotion-4"
    >
      <div
        className="emotion-5"
      >
        <table
          className="emotion-6"
        >
          <thead>
            <tr>
              <th
                className="emotion-7"
              >
                <div
                  className="emotion-8"
                >
                  Name
                </div>
              </th>
              <th
                className="emotion-7"
              >
                <div
                  className="emotion-8"
                >
                  Type
                </div>
              </th>
              <th
                className="emotion-11"
              >
                <div
                  className="emotion-12"
                >
                  <svg
                    data-file-name="kebab.svg"
                    height={20}
                  />
                </div>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              className="emotion-13"
            >
              <td
                colSpan={3}
              >
                <div
                  className="emotion-14"
                >
                  <div
                    className="emotion-15"
                  >
                    <svg
                      data-file-name="noJobs.svg"
                      width={200}
                    />
                    There are currently no models associated with this dataset.
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
